server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -156,22 +168,18 @@ public SearchHit(StreamInput in) throws IOException {
         if (in.readBoolean()) {
             explanation = readExplanation(in);
         }
-        int size = in.readVInt();
-        if (size == 0) {
-            fields = emptyMap();
-        } else if (size == 1) {
-            DocumentField hitField = DocumentField.readDocumentField(in);
-            fields = singletonMap(hitField.getName(), hitField);
+        if (in.getVersion().onOrAfter(Version.V_8_0_0)) {
```

Comment:
This should be 7.3.0, since that is currently 7.x, where this would be backported to.

Reply:
in the last rebasing I adapted to 8.0.0 again; @mayya-sharipova should I maybe use an earlier version instead, e.g. 7.7.0?